### PR TITLE
Homework 04 11 2024

### PR DESCRIPTION
**Pull request by HKV203**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,4 +35,9 @@ build/
...
 
 ### Mac OS ###
-.DS_Store
\ No newline at end of file
+.DS_Store
+/.idea/encodings.xml
+/.idea/misc.xml
+/.idea/uiDesigner.xml
+/.idea/vcs.xml
+/.idea/workspace.xml
```
Comment (Alexmansar) [inline .gitignore:40]: could be just .dea folder)

File: pom.xml [hunk 2]
```diff
@@ -7,11 +7,60 @@
...
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+        </plugin>
+    </plugins>
+</build>
```
Comment (Alexmansar) [inline pom.xml]: move it in the end of pom

File: src/test/java/com/andersen/bookstore/BookTests.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.andersen.bookstore;
+
+import com.andersen.bookstore.model.Book;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class BookTests {
```
Comment (Alexmansar) [inline src/test/java/com/andersen/bookstore/BookTests.java]: BookTest

File: src/test/java/com/andersen/bookstore/BookTests.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Test
+    public void GetTitle_ReturnTestTitle(){
+        assertEquals(book.getTitle(),title);
+    }
+
+    @Test
+    public void GetAuthor_ReturnTestAuthor()
+    {
```
Comment (Alexmansar) [inline src/test/java/com/andersen/bookstore/BookTests.java]: move on 30 line

File: src/test/java/com/andersen/bookstore/BookTests.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        assertEquals(book.getAuthor(),author);
+    }
+    @Test
+    public void GetPrice_ReturnTestPrice()
```
Comment (Alexmansar) [inline src/test/java/com/andersen/bookstore/BookTests.java]: same
and add empty line please

File: src/test/java/com/andersen/bookstore/BookstoreTests.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import com.andersen.bookstore.model.Bookstore;
+import com.andersen.bookstore.model.Order;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class BookstoreTests {
```
Comment (Alexmansar) [inline src/test/java/com/andersen/bookstore/BookstoreTests.java]: ...Test

File: src/test/java/com/andersen/bookstore/BookstoreTests.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    @BeforeEach()
+    public void setup() {
+        List<Order> orders = new LinkedList<>();
+        List<Book> books = new LinkedList<>(List.of(new Book("1", "2", 3)));
+        bookstore = new Bookstore(books, orders);
+        bookstore.openOrder();
+    }
+
+    @Test
+    public void OpenOrder_AddNewOrder_CurrentOrderNotNull() {
```
Comment (Alexmansar) [inline src/test/java/com/andersen/bookstore/BookstoreTests.java]: openOrde_addNew....

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@Mopgonpblr 
Hi) some general comments. Please fix it
And what about CI/CD?

**Comment by HKV203** [comment]
> @Mopgonpblr Hi) some general comments. Please fix it And what about CI/CD?

@Alexmansar  I've tried to implement pull request workflow into the main branch but I didn't get why it didn't work

File: src/test/java/com/andersen/bookstore/BookTests.java (hunk 4 above)
Comment (Alexmansar) [reply src/test/java/com/andersen/bookstore/BookTests.java]: @Mopgonpblr  and this?

**Comment by Alexmansar** [comment]
@Mopgonpblr but in main you don't have tests) try to fix conflicts as well

**Review by Alexmansar — APPROVED** [review]
@Mopgonpblr Please, merge it

**Pull request merged**